### PR TITLE
feat(console,core): remove custom token claims api context dev guard

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Remove custom token claims api context dev guard.

## Checklist

- [x] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: .changeset/spicy-cameras-sleep.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+"@logto/core": minor
+---
+
+add access deny method to the custom token claims script
+
+Introduce a new `api` parameter to the custom token claims script. This parameter is used to provide more access control context over the token exchange process.
+Use `api.denyAccess()` to reject the token exchange request. Use this method to implement your own access control logics.
+
+```javascript
+const getCustomJwtClaims: async ({api}) => {
```
Comment (gao-sun) [inline .changeset/spicy-cameras-sleep.md:12]:
```suggestion
const getCustomJwtClaims: async ({ api }) => {
```

File: .changeset/spicy-cameras-sleep.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  // Reject the token exchange request
```
Comment (gao-sun) [inline .changeset/spicy-cameras-sleep.md:13]: it's not for token exchange only?

File: .changeset/spicy-cameras-sleep.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  api.denyAccess('Access denied');
```
Comment (gao-sun) [inline .changeset/spicy-cameras-sleep.md:14]: the comment better to say the message is customizable

Comment (gao-sun) [reply .changeset/spicy-cameras-sleep.md:14]: nit:
```suggestion
  return api.denyAccess('Access denied');
```

**Review by gao-sun — APPROVED** [review]

File: .changeset/spicy-cameras-sleep.md (hunk 3 above)
Comment (simeng-li) [reply .changeset/spicy-cameras-sleep.md:14]: the return type of `getCustomJwtClaims` must be a JSON object.

**Pull request merged**